src/app/new-page/component/content-multiselect-option/content-multiselect-option.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
+import {
+  Content,
+  MultiselectOption
+} from 'src/app/services/xml-parser-service/xmp-parser.service';
+import { PageService } from '../../service/page-service.service';
+
+@Component({
+  selector: 'app-content-new-multiselect-option',
+  templateUrl: './content-multiselect-option.component.html',
+  styleUrls: ['./content-multiselect-option.component.css']
+})
+export class ContentMultiselectOptionNewComponent implements OnChanges {
+  // eslint-disable-next-line @angular-eslint/no-input-rename
+  @Input() item: MultiselectOption;
+  option: MultiselectOption;
+  contents: Content[];
+  selected: boolean;
+  ready: boolean;
+  state: any;
+
+  constructor(private pageService: PageService) {
+    this.state = this.pageService.parserState();
+  }
+
+  ngOnChanges(changes: SimpleChanges) {
+    for (const propName in changes) {
+      if (changes.hasOwnProperty(propName)) {
+        switch (propName) {
+          case 'item': {
+            if (
+              !changes['item'].previousValue ||
+              changes['item'].currentValue !== changes['item'].previousValue
+            ) {
+              this.ready = false;
+              this.option = this.item;
+              this.init();
+            }
+          }
+        }
+      }
+    }
+  }
+
+  onClick() {
+    this.option.toggleSelected(this.state);
+  }
+
+  private init(): void {
+    this.option.watchIsSelected(this.state, (value) => (this.selected = value));
```

Comment:
this returns a `FlowWatcher` object that has a `close()` method that should be called when you no longer want notifications about the selected state.

You will probably want to capture this in class property that can be properly disposed of when this component is destroyed